Update changelog for v12.0.1 (#26348)
MetaMask/metamask-extension: CHANGELOG.md

## [Unreleased]

## [12.0.1]

> ### Changed
> - Disabled the "Restore Backup" feature ([#26338](https://github.com/MetaMask/metamask-extension/pull/26338))

Reviewer: Nit "removed" seems like a better description, removed is for removed features. I added a bit more detail on the reasoning as well.
```suggestion
### Removed
- Disabled the "Restore Backup" feature ([#26338](https://github.com/MetaMask/metamask-extension/pull/26338))
  - This was linked to reports of state corruption. It has been disabled at least until we can ensure it is safe to use.
```

Reviewer: We should also update this to use the original PR number:

26325

```suggestion
### Removed
- Disabled the "Restore Backup" feature ([#26325](https://github.com/MetaMask/metamask-extension/pull/26325))
  - This was linked to reports of state corruption. It has been disabled at least until we can ensure it is safe to use.
```

### Fixed

> - Fixed corrupted state for some users, which could prevent further data updates from persisting between sessions ([#26330](https://github.com/MetaMask/metamask-extension/pull/26330))

Reviewer: Nit: Listing a few more PRs related to preventing state corruption:
```suggestion
- Fixed corrupted state for some users, which could prevent further data updates from persisting between sessions ([#26330](https://github.com/MetaMask/metamask-extension/pull/26330), [#26305](https://github.com/MetaMask/metamask-extension/pull/26305), [#26338](https://github.com/MetaMask/metamask-extension/pull/26338))
```

Reviewer: Updated to use original PR numbers:

```suggestion
- Fixed corrupted state for some users, which could prevent further data updates from persisting between sessions ([#26325](https://github.com/MetaMask/metamask-extension/pull/26325), [#26280](https://github.com/MetaMask/metamask-extension/pull/26280), [#26265](https://github.com/MetaMask/metamask-extension/pull/26265), [#26282](https://github.com/MetaMask/metamask-extension/pull/26282), [#26298](https://github.com/MetaMask/metamask-extension/pull/26298), [#26302](https://github.com/MetaMask/metamask-extension/pull/26302), [#26308](https://github.com/MetaMask/metamask-extension/pull/26308), [#26310](https://github.com/MetaMask/metamask-extension/pull/26310))
```

## [Unreleased]

## [12.0.1]
### Changed

> - Limit the transaction history in the user's data, and in state logs, to 100 entries ([#26236](https://github.com/MetaMask/metamask-extension/pull/26236))

Reviewer: I would describe this more as a fix, to prevent pending transactions from leading to excessive disk space usage.

The history affected here isn't really user-visible. The minimal use of the history in the activity log is hoped to be mostly unaffected by this change (if it is affected, it would be in a rare edge case). Users would probably think "transaction history" means the list of previous transactions, which is not affected by this change.

- Prevent crashes / error screens when taking some token actions if the "Show balance and token price checker" setting is off ([#26264](https://github.com/MetaMask/metamask-extension/pull/26264))
- Ensure error messages are correctly shown to the user inline on the send screen, if an error occurs when attempting to create the transaction ([#26253](https://github.com/MetaMask/metamask-extension/pull/26253))
- Fix error that could prevent the send screen from loading ([#26295](https://github.com/MetaMask/metamask-extension/pull/26295))

> - Fix crash that could occur some network data is correct or outdated ([#26327](https://github.com/MetaMask/metamask-extension/pull/26327))

Reviewer: This one was missing something grammatically, and probably meant to say "incorrect" rather than "correct". But this might be a better way of describing it anyway: 
```suggestion
- Fix crash that could occur when selected network is missing from network configuration ([#26327](https://github.com/MetaMask/metamask-extension/pull/26327))
```

> - Migrate transaction history data so that they are limited to 100 entries ([#26291](https://github.com/MetaMask/metamask-extension/pull/26291))

Reviewer: The user impact here is that we're fixing a problem with certain transactions taking up an excessive amount of disk space + memory. "Transaction history" as referred to here is not a user-facing feature, and would be confused with the list of previous transactions

- Prevent crashes / error screens when taking some token actions if the "Show balance and token price checker" setting is off ([#26264](https://github.com/MetaMask/metamask-extension/pull/26264))
- Ensure error messages are correctly shown to the user inline on the send screen, if an error occurs when attempting to create the transaction ([#26253](https://github.com/MetaMask/metamask-extension/pull/26253))
- Fix error that could prevent the send screen from loading ([#26295](https://github.com/MetaMask/metamask-extension/pull/26295))
- Fix crash that could occur when selected network is missing from network configuration ([#26327](https://github.com/MetaMask/metamask-extension/pull/26327))
- Migrate transaction history data so that they are limited to 100 entries ([#26291](https://github.com/MetaMask/metamask-extension/pull/26291))

> - Fix crash that could occur some network data is correct or outdated ([#26327](https://github.com/MetaMask/metamask-extension/pull/26327))

Reviewer: Duplicate entry

```suggestion(https://github.com/MetaMask/metamask-extension/pull/26327))
```
